lib/cmdline.c

```diff
@@ -1167,42 +1145,130 @@ static bool rm_cmd_set_cmdline(RmCfg *cfg, int argc, char **argv) {
     return true;
 }
 
-static bool rm_cmd_set_paths(RmSession *session, char **paths) {
-    bool is_prefd = false;
-    bool all_paths_valid = true;
-    bool stdin_paths_preferred = false;
-
-    RmCfg *cfg = session->cfg;
-
-    /* Check the directory to be valid */
-    for(int i = 0; paths && paths[i]; ++i) {
-        if(strcmp(paths[i], "-") == 0) {
-            cfg->read_stdin = true;
+typedef struct rm_cmd_set_paths_vars {
+    RmCfg *const cfg;
+    char **const paths;
+    unsigned int index;
+    bool is_prefd;
+    bool read_stdin;
+    bool stdin_paths_preferred;
+    const bool null_separated;
+    bool all_paths_valid;
+} rm_cmd_set_paths_vars;
+
+static INLINE
+void rm_cmd_set_paths_from_cmdline(
+    rm_cmd_set_paths_vars *const v,
+    const bool replay
+) {
+    g_assert(v);
+    g_assert(v->paths);
+
+    for(char *path, **list = v->paths; (path = *list); ++list) {
+        if(path[0] == '-' && path[1] == 0) {
```

Comment:
Please revert that to ``strcmp`` if you don't have numbers that prove otherwise. That is just over-optimization for me and does not help readability.

Reply:
In addition to [the other comment](https://github.com/sahib/rmlint/pull/319#discussion_r250416132), I do not think there is such a thing as "over-optimization"; the problem with optimization is not that it's over done, but rather as you imply: It might damage some other aspect of the program, such as readability.

Then again, why should readability be the winning aspect? Perhaps `strcmp()` is *over-readablized*, to the unnecessary detriment of optimization. Indeed, a C-string is explicitly an array of characters, and inspecting individual characters (especially when looking for a very short string) is quite normal&mdash;it's one of the killer features C was designed to have.

If there is a need to improve readability, then other means should be employed before sacrificing run-time efficiency: Abstraction facilities (such as macros or inlinable functions or structs), or even just plain, free-form, natural human descriptions in the form of embedded comments.

I mean is something like `strcmp(x,"y") == 0` actually *that* readable? Sure, a competent C programmer should be familiar with it, but he should also be familiar with indexing individual characters, too; both are equally readable from that perspective, so one might as well choose the solution that has better performance.